src/AppWrapper.js

```diff
@@ -0,0 +1,68 @@
+import 'typeface-roboto'
+import 'material-design-icons-iconfont/dist/material-design-icons.css'
+import 'flexboxgrid/css/flexboxgrid.css'
+
+import React from 'react'
+import { Provider } from 'react-redux'
+import { ConnectedRouter } from 'connected-react-router'
+
+import {
+    MuiThemeProvider as Mui3ThemeProvider,
+    createMuiTheme as createMui3Theme,
+} from '@material-ui/core/styles'
+
+import { useConfig } from '@dhis2/app-runtime'
+import { useD2 } from '@dhis2/app-runtime-adapter-d2'
+import { mui3theme } from '@dhis2/d2-ui-core'
+import D2UIApp from '@dhis2/d2-ui-app'
+import { CssReset } from '@dhis2/ui-core'
+
+import { initApi } from './utils/api'
+import history from './utils/history'
+import injectTranslationsToD2 from './utils/injectTranslationsToD2'
+import store from './redux/store'
+import './locales'
+
+import App from './App'
+
+const MUI3Theme = createMui3Theme(mui3theme)
+
+const AppWrapper = () => {
+    const { baseUrl } = useConfig()
+    const { d2 } = useD2({
+        d2Config: {
+            baseUrl: baseUrl + '/api',
+            schemas: [
+                'dataApprovalLevel',
+                'dataSet',
+                'organisationUnit',
+                'organisationUnitGroupSet',
+                'report',
+                'reportTable',
+            ],
+        },
+        onInitialized: d2 => {
+            initApi(d2)
+            injectTranslationsToD2(d2)
+        },
+    })
+
+    if (!d2) {
+        return null
+    }
+
+    return (
+        <D2UIApp>
```

Comment:
Would be nice to remove this, but not critical

Reply:
Some of the old `d2-ui` components require the MUI theme to be on in the context. I think it'd be easier to keep this component and remove it later than to manually add the theme to the context and remove the logic that does that.